features/bootstrap/FeatureContext.php

```diff
@@ -6,39 +6,90 @@
 use WP_CLI\Process;
 use WP_CLI\Utils;
 
-// Inside a community package
-if ( file_exists( __DIR__ . '/utils.php' ) ) {
-	require_once __DIR__ . '/utils.php';
-	require_once __DIR__ . '/Process.php';
-	require_once __DIR__ . '/ProcessRun.php';
-	$project_composer = dirname( dirname( dirname( __FILE__ ) ) ) . '/composer.json';
-	if ( file_exists( $project_composer ) ) {
-		$composer = json_decode( file_get_contents( $project_composer ) );
-		if ( ! empty( $composer->autoload->files ) ) {
-			$contents = 'require:' . PHP_EOL;
-			foreach ( $composer->autoload->files as $file ) {
-				$contents .= '  - ' . dirname( dirname( dirname( __FILE__ ) ) ) . '/' . $file . PHP_EOL;
-			}
-			$folder = sys_get_temp_dir() . '/wp-cli-package-test';
-			if ( is_dir( $folder ) || mkdir( $folder ) || is_dir( $folder ) ) {
-				$project_config = "{$folder}/config.yml";
-				file_put_contents( $project_config, $contents );
-				putenv( 'WP_CLI_CONFIG_PATH=' . $project_config );
-			};
+/**
+ * Load required support files as needed before heading into the Behat context.
+ */
+function wpcli_bootstrap_behat_feature_context() {
+	// We try to detect the vendor folder in the most probable locations.
+	$vendor_locations = [
+		// wp-cli/wp-cli-tests is the root project.
+		dirname( dirname( __DIR__ ) ) . '/vendor',
+		// wp-cli/wp-cli-tests is a dependency.
+		dirname( dirname( dirname( __DIR__ ) ) ),
+	];
+
+	$vendor_folder = '';
+	foreach ( $vendor_locations as $location ) {
+		if ( is_dir( $location )
+		     && is_readable( $location )
+		     && is_file( "{$location}/autoload.php" ) ) {
+			$vendor_folder = $location;
+			break;
 		}
 	}
-	// Inside WP-CLI
-} else {
-	require_once __DIR__ . '/../../php/utils.php';
-	require_once __DIR__ . '/../../php/WP_CLI/Process.php';
-	require_once __DIR__ . '/../../php/WP_CLI/ProcessRun.php';
-	if ( file_exists( __DIR__ . '/../../vendor/autoload.php' ) ) {
-		require_once __DIR__ . '/../../vendor/autoload.php';
-	} elseif ( file_exists( __DIR__ . '/../../../../autoload.php' ) ) {
-		require_once __DIR__ . '/../../../../autoload.php';
+
+	// Didn't manage to detect a valid vendor folder.
+	if ( empty( $vendor_folder ) ) {
+		return;
+	}
+
+	// We assume the vendor folder is located in the project root folder.
+	$project_folder = dirname( $vendor_folder );
+
+	// Now we need to detect the location of wp-cli/wp-cli package.
+	$framework_locations = [
+		// wp-cli/wp-cli is the root project.
+		$project_folder,
+		// wp-cli/wp-cli is a dependency.
+		"{$vendor_folder}/wp-cli/wp-cli",
+	];
+
+	$framework_folder = '';
+	foreach ( $framework_locations as $location ) {
+		if ( is_dir( $location )
+		     && is_readable( $location )
+		     && is_file( "{$location}/php/utils.php" ) ) {
```

Comment:
These two lines have space based precision alignment, please adjust this to ordinary tab indents.